hub/codec/image_codecs.py

```diff
@@ -0,0 +1,338 @@
+            arr = np.reshape(arr, arr.shape[:-1])
+            append_one = True
+        if not self.single_channel or append_one:
+            shape_dims = 2
+        else:
+            shape_dims = 3
+        if len(arr.shape) < shape_dims:
+            raise ValueError(
+                f"The shape length {len(arr.shape)} of the given array should "
+                f"be greater than the number of expected dimensions {shape_dims}"
+            )
+        if len(arr.shape) == shape_dims:
+            return self._msgpack.encode(
+                [
+                    {
+                        "items": self.encode_single_image(arr),
+                        "append_one": append_one,
+                        "image_shape": arr.shape,
+                    }
+                ]
+            )
+        else:
+            image_shape = arr.shape[-shape_dims:]
+            items_shape = arr.shape[:-shape_dims]
+            items = []
+            for i in np.ndindex(items_shape):
+                items.append(self.encode_single_image(arr[i]))
+            return self._msgpack.encode(
+                [
+                    {
+                        "items": items,
+                        "items_shape": items_shape,
+                        "image_shape": image_shape,
+                        "dtype": str(arr.dtype),
+                        "append_one": append_one,
+                    }
+                ]
+            )
+
+    def decode(self, buf: bytes) -> np.ndarray:
+        """
+        Decode images from buffer.
+
+        Example:
+            images_decoded = codec.decode(images_encoded)
+
+        Args:
+            buf (bytes): Encoded images and metadata.
+
+        Returns:
+            Decoded image or array with multiple images.
+        """
+        data = self._msgpack.decode(buf)[0]
+        pass_shape = True if self.__name__ == "webp" else False
+        if "items_shape" not in data:
+            if pass_shape:
+                images = self.decode_single_image(data["items"], data["image_shape"])
+            else:
+                images = self.decode_single_image(data["items"])
+        else:
+            items = data["items"]
+            images = np.zeros(
+                data["items_shape"] + data["image_shape"], dtype=data["dtype"]
+            )
+
+            for i, index in enumerate(np.ndindex(tuple(data["items_shape"]))):
+                if pass_shape:
+                    images[index] = self.decode_single_image(
+                        items[i], data["image_shape"]
+                    )
+                else:
+                    images[index] = self.decode_single_image(items[i])
+
+        if data.get("append_one"):
+            images = np.reshape(images, images.shape + (1,))
+        return images
+
+    def get_config(self):
+        """Get compressor configuration dict"""
+        return {"id": self.codec_id, "single_channel": self.single_channel}
+
+
+class JpegCodec(BaseImgCodec):
```

Comment:
also, can we move all classes into their own separate files? like `JpegCodec` can be in `hub.core.compression.jpeg` and accessible like `from hub.core.compression import JPEG`?